core/services/chainlink/application.go

```diff
@@ -791,16 +791,19 @@ func (app *ChainlinkApplication) GetFeedsService() feeds.Service {
 }
 
 // ReplayFromBlock implements the Application interface.
-func (app *ChainlinkApplication) ReplayFromBlock(chainID *big.Int, number uint64, forceBroadcast bool) error {
+func (app *ChainlinkApplication) ReplayFromBlock(ctx context.Context, chainID *big.Int, number uint64, forceBroadcast bool) error {
 	chain, err := app.Chains.EVM.Get(chainID)
 	if err != nil {
 		return err
 	}
 	chain.LogBroadcaster().ReplayFromBlock(int64(number), forceBroadcast)
 	if app.Config.FeatureLogPoller() {
-		if err := chain.LogPoller().Replay(context.Background(), int64(number)); err != nil {
-			return err
-		}
+		go func() {
+			err := chain.LogPoller().Replay(ctx, int64(number))
+			if err != nil {
+				app.logger.Error(err)
+			}
+		}()
```

Comment:
On second thought, I think I should change this to not be in a separate go routine.

On develop branch, the web server waits until either the replay is complete or HTTP_SERVER_WRITE_TIMEOUT (or whatever the 2.0 analog is?) is reached before giving a response.

In this branch, it's fully async.  It will respond immediately, and the logs will indicate later if the replay was successfully initiated or if there was some problem.

If we call it in a blocking way instead of non-blocking, then the behavior is pretty similar to what's on the develop branch:  web response will wait for either server timeout or completion.  It would be better though, because now we can include a message about whether the replay was successfully started, still in progress, completed, or errored out.  I think this is probably the best solution, as calling it in a separate go routine means we can't actually give the user any information about what's happening... they'll have to check the logs for that, and even then... it may say "in progress" simply because the context was cancelled immediately.  In fact, come to think of it... it could even say "aborted by client" if the web server exits before the main log poller loop has a chance to receive the replay start request.